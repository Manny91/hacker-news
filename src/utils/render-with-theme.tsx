import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "../styled-components";
import { theme } from "../theme";
import "jest-styled-components";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
