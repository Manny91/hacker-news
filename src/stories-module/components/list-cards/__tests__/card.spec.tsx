import React from "react";
import Card from "../card";

import { renderWithTheme } from "../../../../utils/render-with-theme";

describe("Card Component", () => {
  it("should render properly", () => {
    const subject = renderWithTheme(
      <Card value="$510" percentage={50} type="revenue" />
    );
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
