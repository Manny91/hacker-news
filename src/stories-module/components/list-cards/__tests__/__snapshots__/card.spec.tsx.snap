// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render properly 1`] = `
.c0 {
  margin-right: 40px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  list-style-type: none;
}

.c0:last-child {
  margin-right: 0px;
}

.c3 {
  font-size: 20px;
}

.c2 {
  background-color: #14137d;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #f6f6f6;
  border-radius: 7px;
  background-color: #00ffff;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 15px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  margin: 0px;
  font-weight: bold;
  font-size: 22px;
  line-height: normal;
  color: #020202;
}

.c9 {
  margin: 0px;
  font-weight: bold;
  font-size: 22px;
  line-height: normal;
  color: #020202;
  text-transform: uppercase;
  font-weight: 400;
  font-size: 11px;
  color: grey;
}

.c8 {
  height: 10px;
}

.c7 {
  margin-left: 5px;
}

.c1 {
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 7px;
  height: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 20px;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-right: 0px;
    margin-bottom: 20px;
  }

  .c0:last-child {
    margin-bottom: 0px;
  }
}

@media (max-width:992px) {
  .c2 {
    display: none;
  }
}

@media (max-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (max-width:992px) {
  .c4 {
    padding: 7px;
  }
}

<div>
  <li
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="sc-AxirZ c2"
      >
        <span
          aria-label="line-chart"
          class="anticon anticon-line-chart c3"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="line-chart"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M888 792H200V168c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v688c0 4.4 3.6 8 8 8h752c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM305.8 637.7c3.1 3.1 8.1 3.1 11.3 0l138.3-137.6L583 628.5c3.1 3.1 8.2 3.1 11.3 0l275.4-275.3c3.1-3.1 3.1-8.2 0-11.3l-39.6-39.6a8.03 8.03 0 00-11.3 0l-230 229.9L461.4 404a8.03 8.03 0 00-11.3 0L266.3 586.7a8.03 8.03 0 000 11.3l39.5 39.7z"
            />
          </svg>
        </span>
      </div>
      <div
        class="c4"
      >
        <div>
          <div
            class="c5"
          >
            <p
              class="c6"
            >
              $510
            </p>
            <div
              class="c7"
              style="color: rgb(56, 158, 13);"
            >
              <span
                aria-label="arrow-up"
                class="anticon anticon-arrow-up c8"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="arrow-up"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                  />
                </svg>
              </span>
              <b>
                50
                %
              </b>
            </div>
          </div>
          <p
            class="c9"
          >
            revenue
          </p>
        </div>
      </div>
    </div>
  </li>
</div>
`;
